### PR TITLE
create first api route and associated test suite

### PR DESCRIPTION
**Pull request by OwenPortfolio**
Created first api get request for ticket three with 404 error handling and tests.

### CODE REVIEW

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
Your afterAll function in testing should either be inline like your beforeEach, or have a return statement.

Please rename your controllers and model files. Going forward it is best practice to have a different controller file for each 'section' that you are querying. For example, topics.controller.js, topics.model.js, articles.controller.js, articles.model.js.

Make these tweaks and then drop me a dm for approval :)

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const request = require('supertest');
+const app = require('../app');
+const db = require('../db/connection.js');
+const seed = require('../db/seeds/seed');
+const testData = require('../db/data/test-data')
+
+afterAll(() => {
+if(db.end)
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:8]: Line 8 is redundant and not really adding anything here :)

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
You may need to amend your testing, we are looking for an object not an array.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+beforeEach(() => seed(testData))
+
+describe('1. GET /api/topics', () => {
+    test('status: 200, returns an array of topic objects with slug and description properties', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({ body }) => {
+            const { topics } = body;
+            expect(topics).toBeInstanceOf(Array);
```
[the remark this answers is not in the document]
Comment (OwenPortfolio) [reply __tests__/app.test.js:21]: I believe we do want an array, if I'm intepreting the ticket correctly; 

"Responds with:

an array of topic objects, each of which should have the following properties:"

The request itself does kick out an array of three objects.

[the remark this answers is not in the document]
Comment (dappernerddesigns) [reply __tests__/app.test.js:21]: Please read your endpoints.json for examples of what we are expecting here :)

[the remark this answers is not in the document]
Comment (OwenPortfolio) [reply __tests__/app.test.js:21]: Ah, gotcha, an object, with an array, with objects.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            topics.forEach((topics) => {
+                expect.objectContaining({
```
[the remark this answers is not in the document]
Comment (OwenPortfolio) [reply __tests__/app.test.js]: Fixed.

**Review by dappernerddesigns — APPROVED** [review]
Fantastic, :) merge when ready!

**Pull request merged**